journal/src/main/java/io/zeebe/journal/file/SegmentedJournal.java

```diff
@@ -0,0 +1,550 @@
+    writer.deleteAfter(indexExclusive);
+  }
+
+  @Override
+  public void deleteUntil(final long index) {
+    final Map.Entry<Long, JournalSegment> segmentEntry = segments.floorEntry(index);
+    if (segmentEntry != null) {
+      final SortedMap<Long, JournalSegment> compactSegments =
+          segments.headMap(segmentEntry.getValue().index());
+      if (!compactSegments.isEmpty()) {
+        log.debug("{} - Compacting {} segment(s)", name, compactSegments.size());
+        for (final JournalSegment segment : compactSegments.values()) {
+          log.trace("Deleting segment: {}", segment);
+          segment.compactIndex(index);
+          segment.close();
+          segment.delete();
+          journalMetrics.decSegmentCount();
+        }
+        compactSegments.clear();
+      }
+    }
+    resetHead(getFirstSegment().index());
+  }
+
+  @Override
+  public void reset(final long nextIndex) {
+    writer.reset(nextIndex);
+  }
+
+  @Override
+  public long getLastIndex() {
+    return writer.getLastIndex();
+  }
+
+  @Override
+  public long getFirstIndex() {
+    if (!isEmpty()) {
+      return segments.firstEntry().getValue().index();
+    } else {
+      return 0;
+    }
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return writer.getNextIndex() - getFirstSegment().index() == 0;
+  }
+
+  @Override
+  public void flush() {
+    writer.flush();
+  }
+
+  @Override
+  public JournalReader openReader() {
+    final SegmentedJournalReader reader = new SegmentedJournalReader(this);
+    readers.add(reader);
+    return reader;
+  }
+
+  @Override
+  public boolean isOpen() {
+    return open;
+  }
+
+  @Override
+  public void close() {
+    segments
+        .values()
+        .forEach(
+            segment -> {
+              log.debug("Closing segment: {}", segment);
+              segment.close();
+            });
+    currentSegment = null;
+    open = false;
+  }
+
+  /** Opens the segments. */
+  private synchronized void open() {
+    final long startTime = System.currentTimeMillis();
+    // Load existing log segments from disk.
+    for (final JournalSegment segment : loadSegments()) {
+      segments.put(segment.descriptor().index(), segment);
+      journalMetrics.incSegmentCount();
+    }
+
+    // If a segment doesn't already exist, create an initial segment starting at index 1.
+    if (!segments.isEmpty()) {
+      currentSegment = segments.lastEntry().getValue();
+    } else {
+      final JournalSegmentDescriptor descriptor =
+          JournalSegmentDescriptor.builder()
+              .withId(1)
+              .withIndex(1)
+              .withMaxSegmentSize(maxSegmentSize)
+              .build();
+
+      currentSegment = createSegment(descriptor);
+      currentSegment.descriptor().update(System.currentTimeMillis());
+
+      segments.put(1L, currentSegment);
+      journalMetrics.incSegmentCount();
+    }
+    journalMetrics.observeJournalOpenDuration(System.currentTimeMillis() - startTime);
```

Comment:
As far as I know these kind of metrics need to be in seconds, so we need to divide by 1000

Reply:
You are right. I fixed it in the `JournalMetrics`.